### PR TITLE
CAM-415: Fix iOS 10 issues

### PR DESCRIPTION
**Pull request by mikew-personal**
#### Ticket
**Required for Vimeans only**
[CAM-415](https://vimean.atlassian.net/browse/CAM-415)

#### Ticket Summary
Get VimeoUpload compiling for iOS 10

#### Implementation Summary
A lot of NSURL methods now return optional values on iOS 10.  I added code to unwrap the optionals so we can compile on Xcode 8 beta 6.

#### How to Test
N/A

### CODE REVIEW

**Comment by alfiehanssen** [comment]
@mikew-personal 

I'm nervous about adding all of these forced unwraps. Can we convert them to `?` unwraps? Otherwise it seems like we're (1) making the library somewhat unstable and (2) deferring the stabilizing work until later.

**Comment by mikew-personal** [comment]
@alfiehanssen I agree that I'd rather check optionals and fail gracefully, however all of the methods that I updated were already using forced unwraps, so I don't think we're adding any additional instability that wasn't already there.  I assumed we'd rather have failures result in crashes that we can catch vs. undefined behavior.  A lot of the methods that were updated return file paths, so if I modified the methods to return empty NSURL's if we fail to create the path for some reason, it would probably result in a lot of weird upload failures in the clients.

**Comment by alfiehanssen** [comment]
@mikew-personal I don't see any force unwraps in those lines that were edited, can you describe more?

if the upload library crashes, it could adversely impact the archiving of upload state. This could cause lost uploads, stuck uploads, not sure.

**Comment by alfiehanssen** [comment]
I think we'd want to return nil instead of an empty URL and check for nil in the locations where the URL is to be used.

File: VimeoUpload/Descriptor System/DescriptorManagerArchiver.swift [hunk 1]
```diff
@@ -62,7 +62,7 @@ class DescriptorManagerArchiver
         let documentsPath = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)[0]
         var documentsURL = NSURL(string: documentsPath)!
         
-        documentsURL = documentsURL.URLByAppendingPathComponent(name)
+        documentsURL = documentsURL.URLByAppendingPathComponent(name)!
```
Comment (alfiehanssen) [inline VimeoUpload/Descriptor System/DescriptorManagerArchiver.swift]: I see the force unwrap above here

File: VimeoUpload/Descriptor System/KeyedArchiver.swift [hunk 2]
```diff
@@ -59,9 +59,9 @@ class KeyedArchiver: ArchiverProtocol
     {
         var URL = NSURL(string: self.basePath)!
```
Comment (alfiehanssen) [inline VimeoUpload/Descriptor System/KeyedArchiver.swift]: Ah and here too

File: VimeoUpload/UIKit/Model/VIMALAsset.swift [hunk 3]
```diff
@@ -40,7 +40,7 @@ import AVFoundation
     {
         get
         {
-            return self.alAsset.defaultRepresentation().url().absoluteString
```
Comment (alfiehanssen) [inline VimeoUpload/UIKit/Model/VIMALAsset.swift]: I deleted this class is the master branch, probably worth merging master up into v1.0. I should have done this

File: VimeoUpload/Upload/Controllers/VideoDeletionManager.swift [hunk 4]
```diff
@@ -77,8 +77,8 @@ public class VideoDeletionManager: NSObject
         let documentsPath = NSSearchPathForDirectoriesInDomains(.DocumentDirectory, .UserDomainMask, true)[0]
         var documentsURL = NSURL(string: documentsPath)!
         
-        documentsURL = documentsURL.URLByAppendingPathComponent(name)
```
Comment (alfiehanssen) [inline VimeoUpload/Upload/Controllers/VideoDeletionManager.swift]: I see the existing force unwrap

**Comment by alfiehanssen** [comment]
@mikew-personal disregard! I expanded the zones around these changes and see what you mean. 

If you have time consider merging up master into v1.0 as I deleted a number of classes.

**Pull request merged**